service/src/main/kotlin/fi/espoo/evaka/invoicing/service/DraftInvoiceGenerator.kt

```diff
@@ -30,13 +30,32 @@ import java.math.BigDecimal
 import java.math.RoundingMode
 import java.time.DayOfWeek
 import java.time.LocalDate
+import java.time.Month
 import java.time.temporal.ChronoUnit
 import java.util.UUID
 
+enum class InvoiceGenerationLogic {
+    Default,
+    Free,
+    IgnoreAbsences,
+}
+
+interface InvoiceGenerationLogicChooser {
+    fun logicForMonth(year: Int, month: Month, childId: ChildId): InvoiceGenerationLogic
+}
+
+class DefaultInvoiceGenerationLogic : InvoiceGenerationLogicChooser {
+    override fun logicForMonth(year: Int, month: Month, childId: ChildId): InvoiceGenerationLogic =
+        InvoiceGenerationLogic.Default
+}
+
+val defaultInvoiceGenerationLogic = DefaultInvoiceGenerationLogic()
```

Comment:
Why do you have a separate `defaultInvoiceGenerationLogic` variable? If the idea is to not create new objects, then you could use kotlin's singletons instead:

```
object DefaultInvoiceGenerationLogic : InvoiceGenerationLogicChooser { ... }
```

Reply:
That was the idea but I didn't know `object` can be used in this way 🙈. Fixed.